### ISSUE TITLE
Verb commands with only one verb

### ISSUE THREAD

**Issue by mauricemeyer** [gsscoder/commandline#170]
I got an issue with using verb commands with only one option, see the following excerpts:

Definition of verb with supported parameters:
```c#
    [Verb("cp", HelpText = "Copy files and directories to another location")]
    internal class CopyOptions
    {
        [Option('f', "force")]
        internal bool Force { get; set; }
        [Option('l', "LoadRequest")]
        internal bool LoadRequest { get; set; }
        [Value(0, Min = 2, Max = 2)]
        internal IEnumerable<String> SourceTarget { get; set; }
    }
```

I was calling the parser with
```c#
var result =  Parser.Default.ParseArguments<CopyOptions>(args);
```
This left me with the parser not parsing any arguments I passed to the command line after the 
As @gsscoder pointed out to me in an e-mail:
>The overload with just one type, that you're using, handles normal options, since by design library assumes that you'll use this feature whit at least two or more verbs.

To handle only one verb, the following parser overload should be used:
```c#
public ParserResult<object> ParseArguments(string[] args, params Type[] types)
```

Therefore, I changed my implementation to the following
```c#
 ParserResult<Object> result = Parser.Default.ParseArguments(args, typeof(CopyOptions));
```

However, this still didn't do the trick as `Force` and  `LoadRequest` are still always `false` and `SourceTarget` is always `null`.

Additional Info: I'm not using only a copy verb option, but several verb options. The issue is the same and doesn't change if I invoke the parser with 
```c#
ParserResult<Object> result = Parser.Default.ParseArguments(args, typeof(CopyOptions), typeof(option2), typeof(option3), [...]);
```

Thanks already @gsscoder for the fast answer to my initial e-mail, I hope you can help me further.

**Comment by gsscoder** [comment]
Hi @mauricemeyer,
can you post also at least a second class you use as verb command?

In such case you can use standard `ParseArguments<T1,T2...>` verb overload.

```csharp
var result =  Parser.Default.ParseArguments<CopyOptions, Options2, ...>(args)
```

**Comment by gsscoder** [comment]
You must define properties as `public`:
```csharp
[Verb("cp", HelpText = "Copy files and directories to another location")]
internal class CopyOptions {
    [Option('f', "force")]
    public bool Force { get; set; }
    [Option('l', "LoadRequest")]
    public bool LoadRequest { get; set; }
    [Value(0, Min = 2, Max = 2)]
    public IEnumerable<String> SourceTarget { get; set; }
}
```
This is semantically equal to your version, but the Parser operates on `public` properties and reflection API doesn't reason semantically but "literally".

Then use this overload with one verb:
```csharp
var result = Parser.Default.ParseArguments(args, typeof(CopyOptions));
if (result.Value.GetType() == typeof(CopyOptions)) {
    var casted = (CopyOptions)result.Value;
    // consume value here
}
```

Or the overload with more type parameters for 2+ verb commands.

**Note:**
It still exists an issue, since you can specify more than one 2 `SourceTarget`s, but should be filed as another issue (anyway I'll write a unit test for it right now).

_Please check if other things work (excluded what stated in the note)._

**Comment by gsscoder** [comment]
Opened an issue for lack check of Max constraint in such case: https://github.com/gsscoder/commandline/issues/172.

**Comment by mauricemeyer** [comment]
Setting the accessibility for all properties to `public` solved the issue I encountered, thank you very much!

**Comment by gsscoder** [comment]
You're wellcome. I've solved also #172 (in a dev branch), I'm waiting a PR from a collaborator than I'll merge it...

**Comment by gsscoder** [comment]
As a side note, #172 merged. Build from source to benefit from latest changes.
```PowerShell
PS> git clone https://github.com/gsscoder/commandline CommandLine
# output omitted
PS> cd CommandLine
PS> .\packages\FAKE.3.35.1\tools\Fake
# output omitted
PS> dir .\build
    Directory: P:\CommandLine\build
Mode                LastWriteTime     Length Name
----                -------------     ------ ----
d----        23/06/2015     04:56            test
-----        23/06/2015     04:56      91648 CommandLine.dll
-----        23/06/2015     04:56     312832 CommandLine.pdb
-----        23/06/2015     04:56      94730 CommandLine.XML
```
